File: 0015-3sum/0015-3sum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        res=[]
        for i in range(len(nums)):
            left=i+1
            right=len(nums)-1
            while left<right:
                if nums[i]+nums[left]+nums[right]==0:
                    res.append((nums[i],nums[left],nums[right]))
                    left+=1
                    right-=1
                elif nums[i]+nums[left]+nums[right]>0:
                    right-=1
                else:
                    left+=1
        return list(set(res))

File: 0015-3sum/0015-3sum.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Set<ArrayList<Integer>> ans = new HashSet<>();
        Arrays.sort(nums);
        for(int i = 0;i<nums.length-2;i++){
            int left = i+1;
            int right = nums.length-1;
            while(left<right){
                int currentSum = nums[i]+nums[left]+nums[right];
                if(currentSum==0){
                    ans.add(new ArrayList<>(Arrays.asList(nums[i],nums[left],nums[right])));
                }
                if(currentSum<0){
                    left++;
                }else{
                    right--;
                }
            }
        }
        return new ArrayList<>(ans);
    }
}